### ISSUE TITLE
Function names prefixed with '_' fail `function-exists()`

### ISSUE THREAD

**Issue by lunelson** [sass/libsass#877]
Just noticed this, in attempting to use @Hugogiraudel 's SassyJSON library. We probably need a more robust test to see if there are any other character combinations that cause the test to fail.

```scss
@function _test() {
  @return 'hello';
}

@function -test() {
  @return 'hello';
}

@function test() {
  @return 'world';
}

.test {
  out: function-exists('_test');
  out: function-exists('-test');
  out: function-exists('test');
}
```
```css
.test {
  out: false;
  out: true;
  out: true; }
```

**Comment by lunelson** [comment]
Update: this also breaks the `call()` function
```scss
@function _test() {
  @return 'hello';
}

@function -test() {
  @return 'hello';
}

@function test() {
  @return 'world';
}

.call {
  out: call('_test');
  out: call('-test');
  out: call('test');
}
```
```css
.call {
  out: _test();
  out: 'hello';
  out: 'world'; }
```

**Comment by lunelson** [comment]
On further testing, having replaced `_` in function names with `-`, I can also say SassyJSON will not work in libsass until the string-interpolation of lists is fixed, and string-interpolation is consistent with ruby-sass generally

**Comment by HugoGiraudel** [comment]
That's very weird since `_` and `-` are strictly equivalent when used in identifiers in Sass.

**Comment by anlutro** [comment]
If you remove the `@function -test` directive you'll see that `call('-test')` still works. So underscores in `@function` directives are being replaced with dashes internally, but not in `call()` and `function-exists()` calls.

**Comment by lunelson** [comment]
Great catch @anlutro .

**Comment by xzyfer** [comment]
Can anyone confirm whether this `_` normalisation happens for other statement types i.e. mixins, variables

**Comment by xzyfer** [comment]
> Since the named arguments are variable names, underscores and dashes can be used interchangeably.
> http://sass-lang.com/documentation/file.SASS_REFERENCE.html#keyword_arguments

**Comment by xzyfer** [comment]
I've got an initial patch for this behaviour https://github.com/sass/libsass/pull/888

@lunelson / @HugoGiraudel could one of you please create spec(s) address the dash-to-underscore? I imagine this is only one of many cases we handle incorrectly.

**Comment by HugoGiraudel** [comment]
I discussed this with @chriseppstein on Hangouts a couple of days ago. At first, he told me *variables only*, but I pointed out that it works for functions and mixins as well. 

Placeholders being tied to usual CSS identifiers rules are not impacted. Map keys and values can be of any data-type so should not be impacted either.

**Comment by xzyfer** [comment]
I'll address this for mixins, functions and variables for now.

**Comment by xzyfer** [comment]
Spec added https://github.com/sass/sass-spec/pull/258.

**Comment by xzyfer** [comment]
This is fixed and will be 3.2.
